src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -640,7 +650,7 @@ public byte[] Http2AltSvcOriginUri
                         Trace("Discarding downgraded HTTP/1.1 connection because connection limit is exceeded");
                     }
 
-                    sslStream.Close();
+                    await connection!.CloseAsync(ConnectionCloseMethod.Immediate, cancellationToken).ConfigureAwait(false);
```

Comment:
Should this be graceful shutdown? We should send TLS alert IMHO.

Reply:
I don't know enough about TLS; I'll make it graceful.